### ISSUE TITLE
Can't access the replay data (process crashed)

### ISSUE THREAD

**Issue by slavik81** [apitrace/apitrace#332]
In qapitrace, after running a replay, I can't access the replay data. I see a brief "Looking up state" followed by "Process crashed". On the command line, I see this error:

```
json_parser - syntax error found,  forcing abort, Line 1 Column 0
```

I checked with git bisect, and it seems this problem was introduced in 3904691fae6d8d487cc66e38a52b29563639181e

Steps to Reproduce:
1. Open apitrace with a trace.
2. Trace -> Replay
3. Enable Core Profile
4. Hit Ok
5. Wait for Replay to Finish
6. Double-click on a frame in qapitrace
7. Observe "Process Crashed" in the status bar

System:
Late 2013 MacBook Pro /w Intel Iris Pro Graphics
Ubuntu 14.04
OpenGL version: 3.3 (Core Profile) Mesa 10.1.3
GLSL version: 3.30

**Comment by jrfonseca** [comment]
This is most likely a bug in Mesa.

To narrow down

    $ gdb /path/to/glretrace
    ...
    (gdb) run --core -D NUMBER_OF_THE_CALL_TO_INSPECT /path/to/your/application.trace > /dev/null
    ...
    (gdb) bt

and paste the output of `bt` here.

**Comment by slavik81** [comment]
Thanks. I don't really understand the change that uncovered the problem, but based on where it's crashing, a Mesa issue does seem likely.

```
(gdb) run --core -D 28 viewobj.trace > /dev/null
Starting program: /home/slavik/ws/apitrace/build/glretrace --core -D 28 viewobj.trace > /dev/null
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff3439742 in ?? () from /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
(gdb) bt
#0  0x00007ffff3439742 in ?? () from /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
#1  0x00007ffff343a777 in ?? () from /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
#2  0x0000000000467c2a in glstate::dumpParameters(JSONWriter&, glstate::Context&) ()
#3  0x0000000000435b4c in glstate::dumpCurrentContext(std::ostream&) ()
#4  0x000000000043172a in GLDumper::dumpState(std::ostream&) ()
#5  0x00000000004241bd in retrace::retraceCall(trace::Call*) ()
#6  0x0000000000426053 in retrace::RelayRunner::runLeg(trace::Call*) ()
#7  0x0000000000425f46 in retrace::RelayRunner::runRace() ()
#8  0x00000000004244b1 in retrace::RelayRace::run() ()
#9  0x000000000042466f in retrace::mainLoop() ()
#10 0x0000000000424e3b in main ()
```

Specifically, it crashes in `glstate_params.cpp:18321`:
`glGetIntegerv(GL_ATOMIC_COUNTER_BUFFER_BINDING, &atomic_counter_buffer_binding);`

That appears related to the property mentioned [here](https://github.com/apitrace/apitrace/blob/3904691fae6d8d487cc66e38a52b29563639181e/retrace/glstate_params.py#L513):
```
# FIXME: Causes Mesa to segfault.  Disable for now.
#('GL_ATOMIC_COUNTER_BUFFER', 'GL_ATOMIC_COUNTER_BUFFER_BINDING', 'GL_MAX_ATOMIC_COUNTER_BUFFER_BINDINGS'),
```

**Comment by jrfonseca** [comment]
Thanks.  I think you're spot on.

This change should avoid the crash when dumping state, without causing loss of information when tracing:

    diff --git a/specs/glparams.py b/specs/glparams.py
    index 170b06f..08352f6 100644
    --- a/specs/glparams.py
    +++ b/specs/glparams.py
    @@ -3359,7 +3359,8 @@ parameters = [
         ("",	X,	1,	"GL_INVERT_OVG_NV"),	# 0x92B4
         ("",	X,	1,	"GL_PRIMITIVE_BOUNDING_BOX_EXT"),	# 0x92BE
         ("",	X,	1,	"GL_ATOMIC_COUNTER_BUFFER"),	# 0x92C0
    -    ("glGet,glGet_i",	I,	1,	"GL_ATOMIC_COUNTER_BUFFER_BINDING"),	# 0x92C1
    +    # FIXME: Causes Mesa to segfault.  Disable for now.
    +    ("_glGet,_glGet_i",	I,	1,	"GL_ATOMIC_COUNTER_BUFFER_BINDING"),	# 0x92C1
         ("glGet_i",	I,	1,	"GL_ATOMIC_COUNTER_BUFFER_START"),	# 0x92C2
         ("glGet_i",	I,	1,	"GL_ATOMIC_COUNTER_BUFFER_SIZE"),	# 0x92C3
         ("",	X,	1,	"GL_ATOMIC_COUNTER_BUFFER_DATA_SIZE"),	# 0x92C4

**Comment by jrfonseca** [comment]
Should be fixed now.

Upstream Mesa should no longer crash as of http://cgit.freedesktop.org/mesa/mesa/commit/?id=eb58aa9cf015e79a0fcf2e088676e6aa1d5dabce

Sooner or later we'll have to back out this workaround, so that we can inspect the value of this state.

**Comment by slavik81** [comment]
I can confirm that this bug is fixed by the workaround in 1b1d86462d3150dceacff927536ded9fcc168419.

Additionally, I patched the my Ubuntu 14.04 version of mesa with [your change](http://cgit.freedesktop.org/mesa/mesa/commit/?id=eb58aa9cf015e79a0fcf2e088676e6aa1d5dabce). I can confirm that apitrace 3904691fae6d8d487cc66e38a52b29563639181e  no longer crashes.
